packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -1224,22 +1224,40 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       this.deleteExistingHostedUIProviderCustomResource();
     }
 
-    if (props.hostedUIProviderCreds) {
-      JSON.parse(props.hostedUIProviderMeta).forEach(this.createHostedUIProviderResource);
+    if (props.hostedUIProviderMeta && props.hostedUIProviderCreds) {
+      const creds = JSON.parse(props.hostedUIProviderCreds);
+
+      JSON.parse(props.hostedUIProviderMeta).forEach((provider: ProviderMeta) => {
+        const providerCreds: ProviderCreds = creds.find(({ ProviderName }: ProviderCreds) => ProviderName === provider.ProviderName);
+        const hasProviderCreds = providerCreds.client_id && (providerCreds.client_secret || providerCreds.private_key);
+
+        this.createHostedUIProviderResource(provider, !!hasProviderCreds);
+      });
     }
   };
 
-  createHostedUIProviderResource = ({ ProviderName, AttributeMapping }: ProviderMeta) => {
+  createHostedUIProviderResource = ({ ProviderName, AttributeMapping }: ProviderMeta, hasCreds: boolean) => {
     const providerDetails =
       ProviderName === SignInWithApple ? this.createHostedUIAppleProviderDetails() : this.createHostedUIProviderDetails(ProviderName);
 
-    const resourceParams: cognito.CfnUserPoolIdentityProviderProps = {
-      attributeMapping: AttributeMapping,
-      providerDetails,
-      providerName: ProviderName,
-      providerType: ProviderName,
-      userPoolId: cdk.Fn.ref('UserPool'),
-    };
+    let resourceParams: cognito.CfnUserPoolIdentityProviderProps;
+
+    if (hasCreds) {
+      resourceParams = {
+        attributeMapping: AttributeMapping,
+        providerDetails,
+        providerName: ProviderName,
+        providerType: ProviderName,
+        userPoolId: cdk.Fn.ref('UserPool'),
+      };
+    } else {
+      resourceParams = {
+        attributeMapping: AttributeMapping,
+        providerName: ProviderName,
+        providerType: ProviderName,
+        userPoolId: cdk.Fn.ref('UserPool'),
+      };
+    }
```

Comment:
```suggestion
    const resourceParams = {
      attributeMapping: AttributeMapping,
      providerName: ProviderName,
      providerType: ProviderName,
      userPoolId: cdk.Fn.ref('UserPool'),
    };
    if (hasCreds) {
      resourceParams.providerDetails = providerDetails;
    } 
```

Reply:
This doesn't work because once the object is instantiated, `providerDetails` becomes read only.

Reply:
You could do something like:
```
    const resourceParams: cognito.CfnUserPoolIdentityProviderProps = {
      attributeMapping: AttributeMapping,
      providerName: ProviderName,
      providerType: ProviderName,
      userPoolId: cdk.Fn.ref('UserPool'),
      providerDetails: hasCreds ? providerDetails : undefined,
    };
```

Though I detest ternaries personally. 

You could also do:

```

    const additionalParams: { [key: string]: unknown } = {};

    if (hasCreds) {
      additionalParams.providerDetails = providerDetails;
    }

    const resourceParams: cognito.CfnUserPoolIdentityProviderProps = {
      attributeMapping: AttributeMapping,
      providerName: ProviderName,
      providerType: ProviderName,
      userPoolId: cdk.Fn.ref('UserPool'),
      ...additionalParams,
    };
```

It's not particularly evil to leave it as is, though. Just a bit verbose and a few repeated lines.

Reply:
I went with your second recommendation - thanks!